src/viking.js

```diff
@@ -1,11 +1,100 @@
 // Soldier
-class Soldier {}
+class Soldier {
+    constructor (health, strength){
+        this.health = health;
+        this.strength = strength;
+    }
+    attack () {
+       return this.strength;
+    }
+    receiveDamage (theDamage){
+         this.health -= theDamage;
+             
+    }   
+}
 
 // Viking
-class Viking {}
+class Viking extends Soldier{
+    constructor(name, health, strength){
+        super(health, strength);
+        this.name = name;
+    }
+    receiveDamage(theDamage){
+        
+        this.health -= theDamage;
+        if (this.health > 0){
+            return `${this.name} has received ${theDamage} points of damage`;
+        }
+        else ( this.health <= 0)
+            return `${this.name} has died in act of combat`; 
+    }
+    battleCry(){
+        return "Odin Owns You All!"
+    }
+}
 
 // Saxon
-class Saxon {}
+class Saxon extends Soldier {
+    constructor(health, strength){
+        super(health, strength);
+    }
+    
+    receiveDamage(theDamage){
+       
 
+        this.health -= theDamage;
+        if (this.health > 0){ 
+        return `A Saxon has received ${theDamage} points of damage`;
+        }
+        else (this.health <= 0) 
+        return `A Saxon has died in combat`;  
+    }
+
+}
+// aqui me dui errores el jasmin .
+/*
 // War
-class War {}
+class War {
+    constructor(){
+        this.vikingArmy = [];
+        this.saxonArmy = [];
+    }
+    addViking(viking){
+        this.vikingArmy.push(viking);
+
+    }
+    addSaxon(saxon){
+        this.saxonArmy.push(saxon);
+    }
+    vikingAttack(){
+        const randomSaxon = this.saxonArmy [Math.floor](Math.random()) * this.saxonArmy.length;
+        const randomViking = this.vikingArmy [Math.floor](Math.random()) * this.vikingArmy.length;
```

Comment:
Aquí tienes un error de sintaxis.
Tienes:
`this.saxonArmy [Math.floor](Math.random()) * this.saxonArmy.length`
Vamos a ir por partes, lo primero que hacemos es acceder al array de saxonsArmy, `this.saxomArmy`. Ese paso perfecto. Ahora lo que queremos es acceder a un elemento (luego veremos para que este elemento sea aleatorio), la forma correcta de acceder es efectivamente indicarle una posición `this.saxonArmy[NUMBER]`, pero si nos fijamos en lo que escribiste, lo que estás haciendo es pasarle una función, siendo más específico la función de redondeo `Math.floor` (sin ejecutar) `this.saxonArmy [Math.floor]`, por tanto esto ya nos devolvería `undefined`.
El cierre de corchete que tienes debería de ir al final: 
`const randomSaxon = this.saxonArmy[Math.floor(Math.random()) * this.saxonArmy.length];`
Y así sería correcto